packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -0,0 +1,59 @@
+import { isElementOfType } from '@deephaven/react-hooks';
+import React, { ReactNode, useMemo, useState } from 'react';
+import {
+  CheckboxGroup as SpectrumCheckboxGroup,
+  SpectrumCheckboxGroupProps,
+} from '@adobe/react-spectrum';
+import Checkbox from '../Checkbox';
+
+export type CheckboxGroupProps = {
+  children: ReactNode;
+} & Omit<SpectrumCheckboxGroupProps, 'children'>;
+
+/**
+ * Augmented version of the Spectrum CheckboxGroup component that supports
+ * primitive item children.
+ */
+export function CheckboxGroup({
+  children,
+  ...props
+}: CheckboxGroupProps): JSX.Element {
+  const [checkedState, setCheckedState] = useState<{ [key: number]: boolean }>(
```

Comment:
Why do we need to store checkbox state?

Reply:
When I use the DH Checkbox, there is this weird effect where I had to store state, since when I was checking one of the checkbox's in the group, all of the checkboxes were going into a checked state.

Reply:
So is this unnecessary if you switch to `SpectrumCheckbox` ?

Reply:
I think something like this will work:
```typescript
const wrappedChildren = useMemo(
  () =>
    ensureArray(children).map(child =>
      isElementOfType(child, Checkbox) ? (
        child
      ) : (
        <Checkbox key={String(child)} value={String(child)}>
          {String(child)}
        </Checkbox>
      )
    ),
  [children]
);
```